### PR TITLE
Realtime Index Task test

### PR DESCRIPTION
**Pull request by pjain1**
Test Realtime Index Task by adding it to the TaskQueue- 
- Expect it to run the task
- Expect task to connect to firehose and initialize plumber
- Announce a Sink for the events
- Ingest all the events from firehose
- Persist and merge all the events
- Publish and push the segment
- Tasks wait for hand off condition to happen as simulation of historical announcing the published segment is not done

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

File: indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java [hunk 1]
```diff
@@ -593,6 +644,77 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
...
+    RealtimeIndexTask realtimeIndexTask = new RealtimeIndexTask(
+        taskId,
+        new TaskResource(taskId, 1),
+        fireDepartment
+    );
+    long startTime = System.currentTimeMillis();
+    tq.add(realtimeIndexTask);
+    //wait for task to process events and publish segment
+    while(mdc.getPublished().size() != 1){
+      if(System.currentTimeMillis() > startTime + 1000){
```
Comment (drcrallen) [inline indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java]: Is it possible to put the timeout at the test level? `@Test(timeout=XXX)` ?

File: indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java [hunk 2]
```diff
@@ -593,6 +644,77 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
...
+        throw new ISE("Segment not yet published ?");
+      }
+      Thread.sleep(100);
```
Comment (nishantmonu51) [inline indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java]: can we use Countdown latch instead of sleep here ?

Comment (pjain1) [inline indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java]: sure. done

File: indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java (hunk 1 above)
Comment (pjain1) [inline indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java]: Timing out on Countdown latch instead.

**Comment by himanshug** [comment]
LGTM

**Pull request merged**